### PR TITLE
Feature/integrate api database

### PR DESCRIPTION
**Pull request by sanglevinh**

### CODE REVIEW

File: frontend/src/pages/Database/Cells.tsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+import { useSelector } from 'react-redux'
+import { selectCurrentUser } from 'store/slice/User/UserSelector'
 
 const Cells = () => {
+  const user = useSelector(selectCurrentUser)
+
   return (
     <DatabaseWrapper>
-      <DatabaseCells />
+      <DatabaseCells user={user}/>
```
[the remark this answers is not in the document]
Comment (sanglevinh) [reply frontend/src/pages/Database/Cells.tsx:11]: > When displaying Cells View, it appears that a similar API is issued twice, is this the expected implementation? (*The same applies to Experiments View)
> 
> * Example
>   
>   * http://localhost:8000/expdb/experiments?offset=0&limit=50&sort=&sort=
>   * http://localhost:8000/expdb/experiments?offset=0&limit=50

I  have checked it.
It seems to be a bug with useEffect when I change the parameters in the router.
I changed the code in "https://github.com/arayabrain/optinist-for-server/pull/64/files#diff-2cdb1bd681169769aefb60edc1885b157e4fd91836dc6d90b2ce3e8f941c1d75R111-R123" it is resolved.
Similar for Experiments

Comment (itutu-tienday) [inline frontend/src/pages/Database/Cells.tsx:11]: When transitioning from the Experiments View to the Cells View, it is assumed that the Cells View will be filtered by Experiment ID.

However, the current API specification does not support searching by Experiment ID (string), so the API specification will be updated.

Comment (itutu-tienday) [inline frontend/src/pages/Database/Cells.tsx:11]: When Cells View is displayed, Cells data is not displayed.

In the API response, Cells data is acquired.
In Experiments View, Experiments data is successfully displayed.

image

**Review by itutu-tienday — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Database/Cells.tsx (hunk 1 above)
Comment (itutu-tienday) [reply frontend/src/pages/Database/Cells.tsx:11]: Add dummy data responses to the GET experiments,cells API. (already pushed to PR branch)

The above seems to have enabled the data to be displayed in the Cell View, but there seems to be a shortcoming in the implementation.

image

File: frontend/src/components/Database/DatabaseExperiments.tsx [hunk 2]
```diff
@@ -457,189 +187,294 @@ const PopupAttributes = ({
...
+  const columnsTable = [
+    ...columns(handleOpenAttributes, handleOpenDialog, cellPath, navigate),
+    ...getColumns,
+  ].filter(Boolean) as GridEnrichedColDef[]
 
   return (
     <DatabaseExperimentsWrapper>
-      <DataGrid
-        {...datas}
+      <DataGridPro
```
Comment (itutu-tienday) [inline frontend/src/components/Database/DatabaseExperiments.tsx:410]: The column headers do not seem to be displayed in the following places.

image

**Review by itutu-tienday — CHANGES_REQUESTED** [review]

File: frontend/src/components/Database/DatabaseExperiments.tsx (hunk 2 above)
Comment (sanglevinh) [reply frontend/src/components/Database/DatabaseExperiments.tsx:410]: > The column headers do not seem to be displayed in the following places.
> 
> image

I check it with design. I think these two columns will not exist title

image

Comment (sanglevinh) [reply frontend/src/components/Database/DatabaseExperiments.tsx:410]: > The column headers do not seem to be displayed in the following places.
> 
> image

I checked the api response. 
I see the header.graph_titles field is empty.
We are using header.graph_titles to display the column when header.graph_titles is empty the column is not visible

Screen Shot 2023-07-25 at 10 39 41 AM

File: frontend/src/pages/Database/Cells.tsx (hunk 1 above)
Comment (sanglevinh) [reply frontend/src/pages/Database/Cells.tsx:11]: > Add dummy data responses to the GET experiments,cells API. (already pushed to PR branch)
> 
> The above seems to have enabled the data to be displayed in the Cell View, but there seems to be a shortcoming in the implementation.
> 
> image

I checked the api response.
I see the header.graph_titles field is empty.
We are using header.graph_titles to display the column when header.graph_titles is empty the column is not visible

File: frontend/src/components/Database/DatabaseExperiments.tsx (hunk 2 above)
Comment (itutu-tienday) [reply frontend/src/components/Database/DatabaseExperiments.tsx:410]: I have an image of the document you are checking here (screen design v1.3)
*The font seems to be white in your environment?

image

File: frontend/src/pages/Database/Cells.tsx (hunk 1 above)
Comment (itutu-tienday) [reply frontend/src/pages/Database/Cells.tsx:11]: As you informed us, it was due to header.graph_titles being from.

However, after setting header.graph_titles and checking the operation, the following problems seem to exist. Please refer to the attached image.

- "Pixel Image" is not displayed
- Graphs images have broken links (dict params embedded in the URL?)

image

**Review by itutu-tienday — CHANGES_REQUESTED** [review]
I will comment reply.

File: frontend/src/components/Database/DatabaseExperiments.tsx (hunk 2 above)
Comment (sanglevinh) [reply frontend/src/components/Database/DatabaseExperiments.tsx:410]: > I have an image of the document you are checking here (screen design v1.3) *The font seems to be white in your environment?
> 
> image


I understand. Now, I will implement it.

Comment (sanglevinh) [reply frontend/src/components/Database/DatabaseExperiments.tsx:398]: > I have an image of the document you are checking here (screen design v1.3) *The font seems to be white in your environment?
> 
> image

I fixed it.

File: frontend/src/pages/Database/Cells.tsx (hunk 1 above)
Comment (sanglevinh) [reply frontend/src/pages/Database/Cells.tsx:11]: > As you informed us, it was due to header.graph_titles being from.
> 
> However, after setting header.graph_titles and checking the operation, the following problems seem to exist. Please refer to the attached image.
> 
> * "Pixel Image" is not displayed
> * Graphs images have broken links (dict params embedded in the URL?)
> 
> image

I fixed it.

**Review by itutu-tienday — APPROVED** [review]
@sanglevinh 
Thanks, I have checked all the noted corrections.

**Pull request merged**